### PR TITLE
[Sentry-issue] Fix property of undefined in text-submission-viewer.component

### PR DESCRIPTION
**Pull request by Olimbia23**
### Checklist

- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context
This PR solved this Sentry Issue: [ARTEMIS-39F](https://sentry.ase.in.tum.de/organizations/artemis/issues/3399/?project=2)

### Description

Error: "Cannot read property 'slice' of undefined"
Now we check if the row variable is defined before calling the insertToken method

### Steps for Testing

I didn't manage to reproduce this problem. It would be great if someone who has touched the Plagiarism part of the codebase can test it. 
Maybe @joschahenningsen you know how to reproduce it?  

### Test Coverage
Didn't change.

### CODE REVIEW

**Review by joschahenningsen — APPROVED** [review]
LGTM

**Review by joschahenningsen — APPROVED** [review]
Ultimately the actual line calculation might be off by one for matches on the first line, but I think your solution should fix the issue :+1:

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts [hunk 1]
```diff
@@ -153,12 +153,16 @@ export class TextSubmissionViewerComponent implements OnChanges {
 
             const idxColumnFrom = from.column - 1 + offsets[idxLineFrom];
 
-            rows[idxLineFrom] = this.insertToken(rows[idxLineFrom], this.tokenStart, idxColumnFrom);
+            if (rows[idxLineFrom]) {
+                rows[idxLineFrom] = this.insertToken(rows[idxLineFrom], this.tokenStart, idxColumnFrom);
+            }
             offsets[idxLineFrom] += this.tokenStart.length;
```
Comment (FrankeLukas) [inline src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts:159]: We probably shouldn't increase the offset if we don't insert anything, or should we? (I'm not sure.)

File: src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts [hunk 2]
```diff
@@ -153,12 +153,16 @@ export class TextSubmissionViewerComponent implements OnChanges {
...
 
             const idxColumnTo = to.column + to.length - 1 + offsets[idxLineTo];
 
-            rows[idxLineTo] = this.insertToken(rows[idxLineTo], this.tokenEnd, idxColumnTo);
+            if (rows[idxLineTo]) {
+                rows[idxLineTo] = this.insertToken(rows[idxLineTo], this.tokenEnd, idxColumnTo);
+            }
             offsets[idxLineTo] += this.tokenEnd.length;
```
Comment (FrankeLukas) [inline src/main/webapp/app/exercises/shared/plagiarism/plagiarism-split-view/text-submission-viewer/text-submission-viewer.component.ts:166]: Same here.

**Review by FuchsDominik — COMMENTED** [review]
Changes look good, agree an Lukas' comment

**Review by krusche — CHANGES_REQUESTED** [review]
I agree: please move the corresponding calls inside the if statement

**Comment by Olimbia23** [comment]
> I agree: please move the corresponding calls inside the if statement

I did it @krusche

**Review by FrankeLukas — APPROVED** [review]

**Review by DanielKainz — APPROVED** [review]
Code changes look good :-)

**Review by coolchock — APPROVED** [review]

**Pull request merged**